pageserver/src/tenant/timeline.rs

```diff
@@ -6511,21 +6559,51 @@ impl Timeline {
             //
             // Caveat: `refresh_gc_info` is in charged of updating the lease map.
             // Here, we do not check for stale leases again.
-            let max_lsn_with_valid_lease = gc_info.leases.last_key_value().map(|(lsn, _)| *lsn);
+            let max_lsn_with_valid_lsn_lease =
+                gc_info.lsn_leases.last_key_value().map(|(lsn, _)| *lsn);
 
             (
                 space_cutoff,
                 time_cutoff,
                 retain_lsns,
-                max_lsn_with_valid_lease,
+                max_lsn_with_valid_lsn_lease,
             )
         };
 
         let mut new_gc_cutoff = space_cutoff.min(time_cutoff.unwrap_or_default());
-        let standby_horizon = self.standby_horizon.load();
+
         // Hold GC for the standby, but as a safety guard do it only within some
         // reasonable lag.
-        if standby_horizon != Lsn::INVALID {
+        // TODO: revisit this once we've fully transitioned to leases. 10GiB isn't _that_ much
+        // of permitted lag at high ingest rates. Yet again, unlimited lag is a DoS risk.
+        // Ideally we implement a monitoring function for RO replica lag in a higher level controller,
+        // e.g. the TBD compute manager aka endpoint controller. It would kill the replica if it's
+        // lagging too much or is otherwise unresponsive. The standby horizon leasing could
+        // then be moved into that controller, i.e., inside our trust domain. Replica can still
+        // advance existing leases, but, control over existence (create/destroy lease) would be
+        // with that controller.
+        // When solving this, solve it generically for lsn leases as well.
+        let min_standby_horizon = self.standby_horizons.min_and_clear_legacy();
+        let flag_evaluation_result = self
+            .feature_resolver
+            .evaluate_multivariate("standby-horizon-which-min")
+            .ok();
+        Span::current().record(
+            "standby_horizon_which_min",
+            tracing::field::debug(&flag_evaluation_result),
+        );
+        let min_standby_horizon = if cfg!(test) || cfg!(feature = "testing") {
+            // TODO: parametrize rust test / test suite over the feature flag?
```

Comment:
You think it's worth doing? I personally don't, it's pretty clear that `legacy` mode is functionally equivalent to what we had before this PR. No need to continue testing it during the transition period.